firebase-sessions/src/main/AndroidManifest.xml

```diff
@@ -18,7 +18,14 @@
   <!--<uses-sdk android:minSdkVersion="16"/>-->
 
   <application>
-    <service android:name="com.google.firebase.components.ComponentDiscoveryService"
+    <provider
+      android:name="com.google.firebase.sessions.provider.FirebaseSessionsProvider"
+      android:authorities="${applicationId}.firebasesessionsprovider"
+      android:directBootAware="true"
+      android:exported="false"
+      android:initOrder="100" />
```

Comment:
I really like this idea. But is there a way we can rely on the [FirebaseInitProvider](https://github.com/firebase/firebase-android-sdk/blob/master/firebase-common/src/main/java/com/google/firebase/provider/FirebaseInitProvider.java) and use that to detect cold star? Are we limited by lack of access to that provider?

Reply:
One of the common ways to setup firebase in a multi-process app is to disable that provider. I don't think we should rely on it. So we can have our own provider, and we're planning on moving some more logic into it.